template.yaml

```diff
@@ -255,7 +271,40 @@ Parameters:
       - "us.anthropic.claude-3-5-sonnet-20241022-v2:0"
       - "us.anthropic.claude-3-7-sonnet-20250219-v1:0"
       - "us.anthropic.claude-sonnet-4-20250514-v1:0"
-    Description: Model to use for Document Analysis Agent (analytics queries). Model access MUST be enabled in Amazon Bedrock.
+    Description: Model to use for Document Analysis Agent (analytics queries, OpenSearch). Model access MUST be enabled in Amazon Bedrock.
+
+  # S3 Vectors Knowledge Base Configuration
+  VectorSimilarityMeasure:
+    Type: String
+    Default: "cosine"
+    AllowedValues:
+      - "cosine"
+      - "euclidean"
+    Description: |
+      Used in S3 Vectors (configurable only at creation time). Euclidean distance compares absolute magnitudes (product prices, ratings). 
+      Cosine similarity finds directional patterns regardless of scale (similar movie tastes, document topics).
+
+  KnowledgeBaseEmbeddingModelId:
+    Type: String
+    Default: "amazon.titan-embed-text-v2:0"
+    AllowedValues:
+      - "amazon.titan-embed-text-v1:0"
+      - "amazon.titan-embed-text-v2:0"
+      - "cohere.embed-english-v3:0"
+      - "cohere.embed-multilingual-v3:0"
+    Description: Bedrock embedding model.
+
+  KnowledgeBaseVectorIndexName:
+    Type: String
+    Default: "default-index"
+    Description: Name of the S3 Vectors index to create and use
+    AllowedPattern: '^[a-zA-Z0-9-_]{1,255}$'
+
+  # S3 Vectors ingestion schedule
+  S3VectorsIngestionScheduleExpression:
```

Comment:
Ingestion schedule should be independenct of vector store being used..  It's related to the data source not the vector store.